ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java

```diff
@@ -568,6 +569,10 @@ private TermIndex shouldNotifyToInstallSnapshot() {
       // should be notified to install the latest snapshot through its
       // State Machine.
       return getRaftLog().getTermIndex(leaderStartIndex);
+    } else if (leaderStartIndex == RaftLog.INVALID_LOG_INDEX) {
```

Comment:
Just out of curiosity: 

so leaderStartIndex == RaftLog.INVALID_LOG_INDEX means the leader has no log at all (all have been purged) thus there is on valid start index?

Reply:
Yes, that is correct.

Reply:
Thanks for catching this case!